### PR TITLE
Get rid of volatile configuration fields for Optaplanner

### PR DESCRIPTION
**Pull request by geoand**
This provides better CDI integration for the Optaplanner beans

### CODE REVIEW

**Comment by geoand** [comment]
cc @ge0ffrey

**Comment by geoand** [comment]
Failing test:

```
[INFO] Quarkus - Integration Tests - Reactive Messaging - AMQP FAILURE
```

has nothing to do with this PR

File: extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/OptaPlannerBeanProvider.java [hunk 1]
```diff
@@ -13,27 +13,18 @@
 
 public class OptaPlannerBeanProvider {
 
-    static volatile SolverConfig solverConfig;
-    static volatile SolverManagerConfig solverManagerConfig;
```
Comment (ge0ffrey) [inline extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/OptaPlannerBeanProvider.java:17]: @geoand I am very happy to see the static fields disappear :) That was ugly!

**Review by ge0ffrey — APPROVED** [review]

File: extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/OptaPlannerBeanProvider.java (hunk 1 above)
Comment (geoand) [reply extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/OptaPlannerBeanProvider.java:17]: Yeah, @mkouba has done some great work in improving the creation of bean from runtime values

**Review by mkouba — APPROVED** [review]

**Pull request merged**